odoo/addons/base/models/ir_mail_server.py

```diff
@@ -248,6 +263,12 @@ def test_smtp_connection(self):
                 (code, repl) = smtp.getreply()
                 if code != 354:
                     raise UserError(_('The server refused the test connection with error %(repl)s', repl=repl))  # noqa: TRY301
+                if autodetect_max_email_size:
+                    max_size = smtp.esmtp_features.get('size')
+                    if not max_size:
+                        raise UserError(_('The server "%(server_name)s" doesn\'t return the maximum email size.',
```

Comment:
Arf, looks like check style is still unhappy about this one.

How about we remove this part:
```py
except UserError:
    raise
```

And modify this part:
```py
except Exception as e:
    if not isinstance(e, UserError):
        _logger.warning("Connection test on %s failed with a generic error.", server, exc_info=True)
        raise UserError(_("Connection Test Failed! Here is what we got instead:\n %s", e)) from e
```

Would that do the trick and achieve the same behavior?